### PR TITLE
Operators documentation

### LINKED ISSUE #7153: Document SQL operators (like `@>` and `~`)
### Is your feature request related to a problem or challenge?

While reviewing https://github.com/apache/arrow-datafusion/pull/6885#pullrequestreview-1554850720 to add support for `@>` and `<@` array containment operators, I found that there is no documentation on the list of supported datafusion operators

### Describe the solution you'd like

I would like documentation for the operators listed in https://github.com/apache/arrow-datafusion/blob/e02ed51b880d09f2156bd192d283aef75cab04b7/datafusion/expr/src/operator.rs#L29-L80

In a new page of the datafusion SQL guide https://arrow.apache.org/datafusion/user-guide/sql/index.html

### Describe alternatives you've considered

_No response_

### Additional context

This would be a great way to get introduced to datafusion, fire up the CLI to test it out, and help the community so marking it as a good first issue

### PR DESCRIPTION
**Pull request by spaydar**
## Which issue does this PR close?

Closes #7153.

## Are these changes tested?

Yes, I can successfully build and view documentation locally as per ["Build & Preview" instructions](https://github.com/apache/arrow-datafusion/blob/main/docs/README.md#build--preview)

### CODE REVIEW

**Review by waynexia — APPROVED** [review]
This document is excellent, appreciate it!

File: docs/source/user-guide/sql/operators.md [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+| Hello, DataFusion!                     |
++----------------------------------------+
+```
+
+### `@>`
+
+Array Contains
+
+```sql
+> SELECT make_array(1,2,3) @> make_array(1,3);
```
Comment (alamb) [inline docs/source/user-guide/sql/operators.md:394]: Should we add the expected output?

```
❯ SELECT make_array(1,2,3) @> make_array(1,3);
+-------------------------------------------------------------------------+
| make_array(Int64(1),Int64(2),Int64(3)) @> make_array(Int64(1),Int64(3)) |
+-------------------------------------------------------------------------+
| true                                                                    |
+-------------------------------------------------------------------------+
```

File: docs/source/user-guide/sql/operators.md [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+```
+
+### `<@`
+
+Array Is Contained By
+
+```sql
+> SELECT make_array(1,3) <@ make_array(1,2,3);
```
Comment (alamb) [inline docs/source/user-guide/sql/operators.md:402]: Also here?

```
❯ SELECT make_array(1,3) <@ make_array(1,2,3);
+-------------------------------------------------------------------------+
| make_array(Int64(1),Int64(3)) <@ make_array(Int64(1),Int64(2),Int64(3)) |
+-------------------------------------------------------------------------+
| true                                                                    |
+-------------------------------------------------------------------------+
```

**Review by alamb — APPROVED** [review]
Thank you so much @spaydar  and @waynexia  -- this looks great


cc @jayzhan211 and @izveigor

File: docs/source/user-guide/sql/operators.md (hunk 1 above)
Comment (spaydar) [reply docs/source/user-guide/sql/operators.md:394]: Thanks @alamb -- added outputs to both examples.

**Comment by alamb** [comment]
Thanks again everyone

**Pull request merged**